src/main/java/com/uid2/admin/vertx/service/ServiceService.java

```diff
@@ -129,6 +129,12 @@ private void handleServiceAdd(RoutingContext rc) {
                 return;
             }
 
+            boolean existsName = serviceProvider.getAllServices().stream().anyMatch(s -> s.getName().equals(name));
```

Comment:
This check could be turned into a function and would make the code look slightly cleaner